### ISSUE TITLE
ALOpsLicenseImport: Exception creating session 

### ISSUE THREAD

**Issue by JellebBE** [HodorNV/ALOps#38]
Using the template the docker container is created/started.
the following error occurs on the step ALOpsLicenseImport:

![image](https://user-images.githubusercontent.com/35337017/66111361-f9e0f400-e5c8-11e9-903b-21a881599774.png)

My Event log shows the following message:
![image](https://user-images.githubusercontent.com/35337017/66111571-71af1e80-e5c9-11e9-844d-78582d4dd3a8.png)

Any advice?

**Comment by waldo1001** [comment]
can you share the yaml / settings of the build pipeline?

**Comment by JellebBE** [comment]
name: $(Build.BuildId)

variables:
- name: 'AppVersion'
  value: '1.0.[yyyyWW].*'
- name: 'dockerimage'
  value: 'mcr.microsoft.com/businesscentral/sandbox:15.0.36590.0-be-ltsc2019'
- name: 'TestFilter'
  value: '50100..50149'
- name: 'bc_license'
  value: 'C:\Licenties\ALOPS.flf'

pool:
  name: Jelleb

steps:
- checkout: self
  clean: true
  
- task: ALOpsDockerStart@1
  inputs:
    docker_image: 'mcr.microsoft.com/businesscentral/sandbox:15.0.36590.0-be-ltsc2019'
    memory_gb: '6'
    container_restart: 'always'

- task: ALOpsDockerWait@1
  inputs:
    search_string: 'Ready for connections!'

- task: ALOpsLicenseImport@1
  inputs:
    usedocker: true 
    license_path: '/TestLicense/ALOPS.flf'
    remove_license_file: false
- task: ALOpsAppCompiler@1
  inputs:
    usedocker: true
    nav_app_version: $(AppVersion)
    failed_on_warnings: true    

- task: ALOpsAppPublish@1
  inputs:
    usedocker: true
    nav_artifact_app_filter: '*.app'   
    skip_verification: true 

- task: ALOpsAppTest@1
  inputs:
    usedocker: true
    import_action: "Skip"
    testfilter: $(TestFilter)
    show_available_tests: true
  continueOnError: true    

- task: PublishTestResults@2
  inputs:
    testResultsFormat: XUnit
    testResultsFiles: '**/TestResults.xml'
    testRunTitle: 'BC Test Results: $(Build.BuildId)'

- task: ALOpsDockerRemove@1
  enabled: true
  condition: always()

**Comment by waldo1001** [comment]
I see you have a variable "bc_license", so may be change that step to:

```
- task: ALOpsLicenseImport@1
  displayName: 'ALOPS License Import'
  inputs:
    usedocker: true
    license_path: $(bc_license)
    remove_license_file: false
```

May be it's just the wrong path?

What we do with license: 
- we create a public download (easy with dropbox / Azure Blob Storage)
- We create a secret variable to the license

I don't know about the docker setup - but if the sets said it successfully created the docker.  You might want tot test this by removing the "ALOpsDockerRemove" step, and check on your agent the condition of your container?

Does the output of your steps look something like this?
![image](https://user-images.githubusercontent.com/12088142/66120940-54378000-e5dc-11e9-8e53-b207401e03e8.png)
basically meaning - did it successfully create and wait for the container?

You might want to remove the "ALOpsAppTest" for now - we're still working on the Wave2 version of running tests...

**Comment by JellebBE** [comment]
I Have changed the variable to a link to my drobox account.(file location) issue still remains.

Result:
![image](https://user-images.githubusercontent.com/35337017/66123137-a16a2080-e5e1-11e9-8f5b-4c4d9d29f552.png)


Following i have tried is to remove the "ALOpsDockerRemove" step.
Resulting:
![image](https://user-images.githubusercontent.com/35337017/66123885-4e916880-e5e3-11e9-959b-7972e9254afe.png)

**Comment by JellebBE** [comment]
Update,

I have just tried it from my home network. that worked without any issue. 
The error must be 'Company Network' related. (Strange message then..)
I will try to find out what the root cause of this is together with our Network team.

**Comment by waldo1001** [comment]
Thanks for the update!  We were really wondering what could be wrong ;-).  You can close the issue if you're ok with that ;-).

**Comment by JellebBE** [comment]
Just an update. The issue was related to the pool agent.  If you run a pool agent on an internal server/computer. Using a self hosted agent. You have to run the Pool Agent as Admin. (it is not mentioned in the MS doc's as far as i have checked, it only mentions that you should run the config in elevated permissions as you want to run the agent as a service.) The error message was not very clear in this case. i had to rethink what i did different (@home) then @ the office. After a while i noticed that i had run the pool agent as admin. So after all its not Network related.

**Comment by waldo1001** [comment]
Good, I like those ;-).
Well, running docker as admin makes sense, right ;-)
